src/ComputationContainer/Random/Csprng.cpp

```diff
@@ -47,23 +49,25 @@ void CSPRNG::GetRand(unsigned char *buf, unsigned int byteSize)
     }
     else
     {
-        unsigned char *seed = new unsigned char[randombytes_SEEDBYTES];
-        syscall(SYS_getrandom, seed, randombytes_SEEDBYTES, 1);
-        randombytes_buf_deterministic(buf, byteSize, seed);
+        std::unique_ptr<unsigned char[]> seed =
+            std::make_unique<unsigned char[]>(randombytes_SEEDBYTES);
+        syscall(SYS_getrandom, seed.get(), randombytes_SEEDBYTES, 1);
+        randombytes_buf_deterministic(buf, byteSize, seed.get());
     }
 };
 
 long long int CSPRNG::GetRandLL()
 {
-    constexpr unsigned int LL_SIZE = sizeof(long long int);  // 8byte = 64bit
+    constexpr std::size_t LL_SIZE = sizeof(long long int);  // 8byte = 64bit
 
-    unsigned char *rnd = new unsigned char[LL_SIZE];
+    // std::make_shared<unsigned char[]> is available in C++20
```

Comment:
Is this comment necessary?
I didn't think it was necessary to mention `make_shared` if `make_unique` was fine.

Reply:
:bow: It is exactly as you said.  
I thought that, in this context, it is prefer to use `std::shared_ptr`.

Reply:
Finally, we used `std::unique_ptr` and ref type.